x/concentrated-liquidity/fees_test.go

```diff
@@ -1015,3 +1023,168 @@ func (s *KeeperTestSuite) TestPreparePositionAccumulator() {
 		})
 	}
 }
+
+type Positions struct {
+	numAccounts    int
+	numFullRange   int
+	numNarrowRange int
+	numConsecutive int
+	numOverlapping int
+}
+
+func (s *KeeperTestSuite) TestFunctionalFees() {
+	positions := Positions{
+		numAccounts:    5,
+		numFullRange:   4,
+		numNarrowRange: 3,
+		numConsecutive: 2,
```

Comment:
idea: we can make a randomized test with various parameters in this struct to rerun multiple times

Reply:
Added issue!

https://github.com/osmosis-labs/osmosis/issues/4361